### PR TITLE
Supplier model

### PR DESCRIPTION
**Pull request by igorsergiichuk**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | no
| Fixed issues      | #

## The problem

Supplier model needed to be added to the project

## How I resolved it

Created new Supplier Model, only uses basic CRUD so no special methods. Test and mocks are in progress.

## How to verify it

Run the tests

Notify the following people: @darrynten

### CODE REVIEW

File: src/Models/Supplier.php [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+     *
+     * @var array $features
+     */
+    protected $features = [
+        'all' => true,
+        'get' => true,
+        'save' => true,
+        'delete' => true,
+    ];
+
```
Comment (darrynten) [inline src/Models/Supplier.php]: This is where StyleCI is failing :)

File: src/Models/Supplier.php [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+ * @link     https://github.com/darrynten/sage-one-php
+ */
+
+namespace DarrynTen\SageOne\Models;
+
+use DarrynTen\SageOne\BaseModel;
+
+/**
+ * Supplier Model
+ *
```
Comment (darrynten) [inline src/Models/Supplier.php:17]: You need to add some information here, and also leave a link to the documentation for this model

**Review by darrynten — CHANGES_REQUESTED** [review]

File: src/Models/Supplier.php (hunk 1 above)
Comment (igorsergiichuk) [reply src/Models/Supplier.php]: I've already fixed that.) Thanks. The only thing I need to pull is review.

**Review by igorsergiichuk — COMMENTED** [review]
Added a link to Supplier parameters:
https://accounting.sageone.co.za/api/1.1.2/Help/Api/POST-Supplier-Save

File: src/Models/Supplier.php [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+ * Details on parameters for Supplier:
+ * https://accounting.sageone.co.za/api/1.1.2/Help/Api/POST-Supplier-Save
```
Comment (darrynten) [inline src/Models/Supplier.php]: This link should be https://accounting.sageone.co.za/api/1.1.2/#bookmark_Supplier

The link you attached for is specific to the save method

**Review by darrynten — CHANGES_REQUESTED** [review]

**Review by mxnr — DISMISSED** [review]
Looks good for me👍

**Comment by darrynten** [comment]
@mxnr it's looking good, we're just waiting on the tests and mocks now :) But I agree, @igorsergiichuk is doing a great job here

File: tests/SageOne/Models/AccountModelTest.php [hunk 4]
```diff
@@ -176,7 +176,7 @@ public function testFeatures()
     public function testGetAll()
     {
         $this->verifyGetAll(Account::class, function ($results) {
-            $this->assertEquals(2, count($results));
+            $this->assertCount(2, $results);
```
Comment (darrynten) [inline tests/SageOne/Models/AccountModelTest.php:179]: Thanks for changing this

File: src/Models/Supplier.php [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            'nullable' => false,
+            'readonly' => false,
+        ],
+        'name' => [
+            'type' => 'string',
+            'nullable' => false,
+            'readonly' => false,
+        ],
+        'category' => [
+            'type' => 'SupplierCategory',
```
Comment (darrynten) [inline src/Models/Supplier.php:48]: Does this model exist yet? If it doesn't then we will need to add it as part of this PR

**Review by darrynten — CHANGES_REQUESTED** [review]

**Comment by darrynten** [comment]
Superseded by #30

**Pull request closed without merging**